image/tarexport/loader.go

```diff
@@ -0,0 +1,128 @@
+package tarexport
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/pkg/progress"
+	digest "github.com/opencontainers/go-digest"
+	ociv1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type loader interface {
+	Load() error
+}
+
+type dockerLoader struct {
+	manifests      []manifestItem
+	tr             *tarexporter
+	tmpDir         string
+	outStream      io.Writer
+	progressOutput progress.Output
+}
+
+func (dl *dockerLoader) Load() error {
+	return dl.tr.loadHelper(dl.manifests, dl.outStream, dl.progressOutput, dl.tmpDir)
+}
+
+type dockerLegacyLoader struct {
```

Comment:
nit: explain what "legacy" means. :)

Reply:
could you actually explain that to me? I've no clue about that code as I wasn't even contributing to docker :)

Reply:
maybe @thaJeztah

Reply:
I think he just means that `legacy` means something new every time you add new code, so let's give this a better name than `LegacyLoader`, maybe based on the spec type/version.

Reply:
ideas actually? I don't remember what was the version there either

Reply:
Maybe `v0TarLoader`?